src/modules/src/Eryph.Modules.VmHostAgent/Networks/OvsDriverProvider.cs

```diff
@@ -0,0 +1,210 @@
+            @$"/l ""{infFileName}"" /c s /i {EryphConstants.DriverModuleName}",
+            infDirectoryPath)
+        from __ in guard(result.ExitCode == 0,
+            Error.New($"Failed to install OVS Hyper-V switch extension:{Environment.NewLine}{result.Output}"))
+        from hostNetworkCommands in default(RT).HostNetworkCommands
+        from ___ in hostNetworkCommands.EnableSwitchExtension()
+        from ____ in logInformation("Successfully installed OVS Hyper-V switch extension {DriverVersion}", infVersion)
+        select unit;
+
+    public static Aff<RT, Unit> uninstallDriver() =>
+        from _ in logInformation("Going to uninstall OVS Hyper-V switch extension...")
+        from hostNetworkCommands in default(RT).HostNetworkCommands
+        from __ in hostNetworkCommands.DisableSwitchExtension()
+        from result in ProcessRunner<RT>.runProcess("netcfg.exe", $"/u {EryphConstants.DriverModuleName}")
+        from ___ in guard(result.ExitCode == 0,
+            Error.New($"Failed to uninstall OVS Hyper-V switch extension:{Environment.NewLine}{result.Output}"))
+        from ____ in logInformation("Successfully uninstalled OVS Hyper-V switch extension")
+        select unit;
+
+    public static Aff<RT, Unit> removeAllDriverPackages() =>
+        from installedDriverPackages in getInstalledDriverPackages()
+        from ___ in installedDriverPackages
+            .Map(di => removeDriverPackage(di.Driver))
+            .TraverseSerial(u => u)
+        select unit;
+
+    internal static Aff<RT, Unit> removeDriverPackage(string infName) =>
+        from _ in logInformation("Going to remove driver package {InfName}...", infName)
+        // The /uninstall flag is not supported on Windows Server 2016
+        from result in ProcessRunner<RT>.runProcess("pnputil.exe", $"/delete-driver {infName} /force")
+        from __ in guard(result.ExitCode == 0,
+            Error.New($"Failed to remove driver package {infName}:{Environment.NewLine}{result.Output}"))
+        from ___ in logInformation("Successfully removed driver package {InfName}", infName)
+        select unit;
+
+    public static Aff<RT, Seq<DismDriverInfo>> getInstalledDriverPackages() =>
+        from psEngine in default(RT).Powershell
+        let command = PsCommandBuilder.Create()
+            .AddCommand("Get-WindowsDriver")
+            .AddParameter("Online")
+        from result in psEngine.GetObjectValuesAsync<DismDriverInfo>(command).ToAff()
+        select result.Map(Optional)
+            .Somes()
+            .Filter(di => di.OriginalFileName?.Contains(
+                 EryphConstants.DriverModuleName, StringComparison.OrdinalIgnoreCase) ?? false);
+
+    public static Aff<RT, bool> isDriverLoaded() =>
+        from result in ProcessRunner<RT>.runProcess("driverquery.exe", "/FO LIST")
+        from match in Eff(() => Regex.Match(
+            result.Output,
+            $@"Module Name:\s*{Regex.Escape(EryphConstants.DriverModuleName)}",
```

Comment:
Looks like driverquery.exe output is language dependent:
On my system the response is different:

PS T:\ovspackage\driver> driverquery.exe /FO LISt | findstr DBO_OVSE
Modulname:                  DBO_OVSE

Searching just for DBO_OVSE would be sufficient, I think

Reply:
Fixed